compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -336,6 +337,88 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Ensures that the last parsed pattern is not followed by a method call or an operator.
+    #[must_use = "the pattern must be discarded as `PatKind::Err` if this function returns Some"]
+    fn maybe_recover_trailing_expr<'b>(
+        &'b mut self,
+        pat: Cow<'b, P<Expr>>,
+        is_end_bound: bool,
```

Comment:
1. Why `pat` is an expression and not a pattern?
2. Can you document `is_end_bound`?

Reply:
The pattern is already parsed, and to parse a potential trailing `+ Expr` I need the left hand, so the pat as an `Expr`.

I've replaced the `pat` param with its span and used `ExprKind::Err` as a dummy since it seems the left hand is only used for the result (e.g. `ExprKind::Try(lhs)`) and not for parsing the next expr.